samples/nrf9160/serial_lte_modem/src/tcpip_proxy/slm_at_udp_proxy.c

```diff
@@ -0,0 +1,608 @@
+		close(udp_sock);
+		return -1;
+	}
+	addr_len = strlen(modem_param.network.ip_address.value_string);
+	if (addr_len == 0) {
+		LOG_ERR("LTE not connected yet");
+		close(udp_sock);
+		return -1;
+	}
+	if (!check_for_ipv4(modem_param.network.ip_address.value_string,
+			addr_len)) {
+		LOG_ERR("Invalid local address");
+		close(udp_sock);
+		return -1;
+	}
+	if (inet_pton(AF_INET, modem_param.network.ip_address.value_string,
+		&local.sin_addr) != 1) {
+		LOG_ERR("Parse local IP address failed: %d", -errno);
+		close(udp_sock);
+		return -EINVAL;
+	}
+
+	ret = bind(udp_sock, (struct sockaddr *)&local,
+		 sizeof(struct sockaddr_in));
+	if (ret) {
+		LOG_ERR("bind() failed: %d", -errno);
+		sprintf(rsp_buf, "#XUDPSVR: %d\r\n", -errno);
+		rsp_send(rsp_buf, strlen(rsp_buf));
+		close(udp_sock);
+		return -errno;
+	}
+
+	udp_thread_id = k_thread_create(&udp_thread, udp_thread_stack,
+			K_THREAD_STACK_SIZEOF(udp_thread_stack),
+			udp_thread_func, NULL, NULL, NULL,
+			THREAD_PRIORITY, K_USER, K_NO_WAIT);
+
+	sprintf(rsp_buf, "#XUDPSVR: %d started\r\n", udp_sock);
+	rsp_send(rsp_buf, strlen(rsp_buf));
+	LOG_DBG("UDP server started");
+
+	return ret;
+}
+
+static int do_udp_server_stop(int error)
+{
+	int ret = 0;
+
+	if (udp_sock > 0) {
+		k_thread_abort(udp_thread_id);
+		ret = close(udp_sock);
+		if (ret < 0) {
+			LOG_WRN("close() failed: %d", -errno);
+			ret = -errno;
+		}
+		udp_sock = INVALID_SOCKET;
+
+		sprintf(rsp_buf, "#XUDPSVR: %d stopped\r\n", error);
+		rsp_send(rsp_buf, strlen(rsp_buf));
+		LOG_DBG("UDP server stopped");
+	}
+
+	return ret;
+}
+
+static int do_udp_client_connect(const char *url, u16_t port, int sec_tag)
+{
+	int ret;
+
+	/* Open socket */
+	if (sec_tag == INVALID_SEC_TAG) {
+		udp_sock = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
+	} else {
+		udp_sock = socket(AF_INET, SOCK_DGRAM, IPPROTO_DTLS_1_2);
+
+	}
+	if (udp_sock < 0) {
+		LOG_ERR("socket() failed: %d", -errno);
+		sprintf(rsp_buf, "#XUDPCLI: %d\r\n", -errno);
+		rsp_send(rsp_buf, strlen(rsp_buf));
+		return -errno;
+	}
+	if (sec_tag != INVALID_SEC_TAG) {
+		sec_tag_t sec_tag_list[1] = { sec_tag };
+
+		ret = setsockopt(udp_sock, SOL_TLS, TLS_SEC_TAG_LIST,
+				sec_tag_list, sizeof(sec_tag_t));
+		if (ret) {
+			LOG_ERR("set tag list failed: %d", -errno);
+			sprintf(rsp_buf, "#XUDPCLI: %d\r\n", -errno);
+			rsp_send(rsp_buf, strlen(rsp_buf));
+			close(udp_sock);
+			return -errno;
+		}
+	}
+
+	/* Connect to remote host */
+	if (check_for_ipv4(url, strlen(url))) {
+		remote.sin_family = AF_INET;
+		remote.sin_port = htons(port);
+		LOG_DBG("IPv4 Address %s", log_strdup(url));
+		/* NOTE inet_pton() returns 1 as success */
+		ret = inet_pton(AF_INET, url, &remote.sin_addr);
+		if (ret != 1) {
+			LOG_ERR("inet_pton() failed: %d", ret);
+			close(udp_sock);
+			return -EINVAL;
+		}
+	} else {
+		struct addrinfo *result;
+		struct addrinfo hints = {
+			.ai_family = AF_INET,
+			.ai_socktype = SOCK_STREAM
```

Comment:
Shouldn't this be `SOCK_DGRAM` as we deal with UDP?

Reply:
Sorry I missed that in previous commit. Now fixed